File: 21-Foreign Function Interface (FFI)/adder.ts
// deno run --allow-ffi adder.ts
const libName = {
  windows: "./adder.dll",
  linux: "./libadder.so",
  darwin: "./libadder.dylib",
}[Deno.build.os];

const lib = Deno.dlopen(libName, {
  add: { parameters: ["i32", "i32"], result: "i32" },
});

const a = 5;
const b = 3;

const result = lib.symbols.add(a, b);
console.log(`Sonuç: ${a} + ${b} = ${result}`);

lib.close();

File: 21-Foreign Function Interface (FFI)/adder.c
// For Linux/MacOS
// gcc -shared -o libadder.so adder.c

// For Window
// gcc -shared -o adder.dll adder.c

#include <stdint.h>

int32_t add(int32_t a, int32_t b) {
    return a + b;
}
